### PR TITLE
Tweaks to build tensorflow-lite-select-tf-ops

### PR DESCRIPTION
**Pull request by jennya**
These were the minor updates I needed in order to successfully build the tensorflow-lite-select-tf-ops.aar dependency with a tflite input model.

Without the protobuf dependency, I could build my app but always got a runtime error when loading the tflite model:

`java.lang.UnsatisfiedLinkError: dlopen failed: cannot locate symbol "_ZNK6google8protobuf7Message11GetTypeNameEv" referenced by "/data/app/org.auderenow.openrdt.reader-2/lib/arm64/libtensorflowlite_flex_jni.so".`

The update to `delegate.cc` is taken directly from the quick fix described here: https://github.com/tensorflow/tensorflow/issues/42744. I'm sure there must be some other fix that would maintain functionality on Windows, but I have no idea what that would be.

### CODE REVIEW

**Comment by jennya** [comment]
@googlebot I signed it!

**Comment by jennya** [comment]
cc @terryheo are there other ways to fix the protobuf issue?

**Comment by terryheo** [comment]
What's the build command you used to see the error?

**Comment by jennya** [comment]
`sh tensorflow/lite/tools/build_aar.sh --input_models=<model.tflite>  --target_archs=arm64-v8a,armeabi-v7a`

**Comment by jennya** [comment]
Or maybe I should be tagging @thaink ?

File: tensorflow/lite/delegates/flex/delegate.cc [hunk 1]
```diff
@@ -149,7 +149,7 @@ extern "C" {
 #if defined(_WIN32)
 __declspec(dllexport)
 #endif
-    tflite::TfLiteDelegateUniquePtr TF_AcquireFlexDelegate() {
-  return tflite::FlexDelegate::Create();
-}
+//    tflite::TfLiteDelegateUniquePtr TF_AcquireFlexDelegate() {
```
Comment (terryheo) [inline tensorflow/lite/delegates/flex/delegate.cc:152]: what happens if you didn't comment this out?

**Comment by terryheo** [comment]
@jennya likely.
Could you confirm that the build command itself doesn't make an error but you'll see the error during runtime?

File: tensorflow/lite/delegates/flex/delegate.cc (hunk 1 above)
Comment (jennya) [reply tensorflow/lite/delegates/flex/delegate.cc:152]: I get this build error:

```
ERROR: /Users/jennyabrahamson/tf/tensorflow/tensorflow/lite/delegates/flex/BUILD:77:1: C++ compilation of rule '//tensorflow/lite/delegates/flex:delegate_only_runtime' failed (Exit 1)
tensorflow/lite/delegates/flex/delegate.cc:152:37: error: 'TF_AcquireFlexDelegate' has C-linkage specified, but returns user-defined type 'tflite::TfLiteDelegateUniquePtr' (aka 'unique_ptr<TfLiteDelegate, void (*)(TfLiteDelegate *)>') which is incompatible with C [-Werror,-Wreturn-type-c-linkage]
    tflite::TfLiteDelegateUniquePtr TF_AcquireFlexDelegate() {
```

**Comment by jennya** [comment]
@terryheo if I don't include the extra protobuf dependency (`clean_dep("@com_google_protobuf//:protobuf"),`), the build succeeds but my android app crashes at runtime when loading the tflite model

File: tensorflow/lite/tools/build_aar.sh [hunk 2]
```diff
@@ -41,7 +41,7 @@ function generate_list_field {
   local list_string="$2"
   local list=(${list_string//,/ })
 
-  local message+=("$name=[")
+  local message=("$name=[")
```
Comment (thaink) [inline tensorflow/lite/tools/build_aar.sh:44]: Thanks for spotting this.
But does this result in build errors?

File: tensorflow/lite/delegates/flex/build_def.bzl [hunk 3]
```diff
@@ -141,6 +141,7 @@ def tflite_flex_cc_library(
                 clean_dep("//tensorflow/core:portable_tensorflow_lib_lite"),
                 clean_dep("//tensorflow/core/platform:strong_hash"),
                 clean_dep("//tensorflow/lite/delegates/flex:portable_images_lib"),
+                clean_dep("@com_google_protobuf//:protobuf"),
```
Comment (thaink) [inline tensorflow/lite/delegates/flex/build_def.bzl:144]: I am fine with this too.

File: tensorflow/lite/tools/build_aar.sh (hunk 2 above)
Comment (jennya) [reply tensorflow/lite/tools/build_aar.sh:44]: Not a build error, my shell wouldn't run it:
`tensorflow/lite/tools/build_aar.sh: line 44: local: `message+': not a valid identifier`

Comment (thaink) [reply tensorflow/lite/tools/build_aar.sh:44]: I see.

File: tensorflow/lite/delegates/flex/build_def.bzl (hunk 3 above)
Comment (thaink) [reply tensorflow/lite/delegates/flex/build_def.bzl:144]: And what happen if you don't add this?
Usually the protos_all_cc will implicitly pull in this dependency.

File: tensorflow/lite/delegates/flex/delegate.cc (hunk 1 above)
Comment (terryheo) [reply tensorflow/lite/delegates/flex/delegate.cc:152]: I'll handle this separately. Could you revert this change and update your PR?

File: tensorflow/lite/delegates/flex/build_def.bzl (hunk 3 above)
Comment (jennya) [reply tensorflow/lite/delegates/flex/build_def.bzl:144]: Without this I get a runtime error when loading the tflite model in my app:

`java.lang.UnsatisfiedLinkError: dlopen failed: cannot locate symbol "_ZNK6google8protobuf7Message11GetTypeNameEv" referenced by "/data/app/org.auderenow.openrdt.reader-2/lib/arm64/libtensorflowlite_flex_jni.so".`

It makes the aar file about 1 MB larger, so would definitely be interested to know if there is a lighter-weight fix

File: tensorflow/lite/delegates/flex/delegate.cc (hunk 1 above)
Comment (jennya) [reply tensorflow/lite/delegates/flex/delegate.cc]: done

File: tensorflow/lite/delegates/flex/build_def.bzl (hunk 3 above)
Comment (thaink) [reply tensorflow/lite/delegates/flex/build_def.bzl:144]: I'll try to reproduce this. Not sure why GetTypeName is not pulled in.

File: tensorflow/lite/delegates/flex/delegate.cc (hunk 1 above)
Comment (terryheo) [reply tensorflow/lite/delegates/flex/delegate.cc]: thanks. BTW, Could you share the NDK version you used?

Comment (jennya) [reply tensorflow/lite/delegates/flex/delegate.cc]: Yeah it's 20.0.5594570

File: tensorflow/lite/delegates/flex/build_def.bzl (hunk 3 above)
Comment (thaink) [reply tensorflow/lite/delegates/flex/build_def.bzl:144]: I tried it but cannot reproduce this issue.
The flex model run fine in my test.

When checking the libtensorflowlite_flex_jni.so with following command I get:
nm -gD libtensorflowlite_flex_jni.so | grep GetTypeName
 U _ZNK6google8protobuf7Message11GetTypeNameEv

Which means GetTypeName is included in the library.

Can you provide more details about your build environment? What tools are you using?

Comment (thaink) [reply tensorflow/lite/delegates/flex/build_def.bzl:144]: Gentle ping on this.

**Comment by gbaned** [comment]
@jennya  As per update from @thaink since there is no reply from you to provide a way to reproduce, we are closing this PR. Thank you!

**Pull request closed without merging**